Todo/AngularJS/TodoFunc.js

```diff
@@ -0,0 +1,88 @@
+var TodoApp = angular.module('todoApp', ['ngMaterial', 'ngMessages'])
+    .config(function ($mdThemingProvider) {
+        $mdThemingProvider.theme('IdoTheme')
+            .primaryPalette('pink', {
+                'default': '900'
+            })
+            .accentPalette('light-blue', {
+                'default': 'A700'
+            })
+            .warnPalette('amber').dark();
+    });
+
+
+
+TodoApp.controller('controller', function ($scope, $http, $mdDialog) {

```

Comment:
create a service for api requests, it shouldnt be in your controller.

it is a lot cleaner and better design-wise